move_base/src/move_base.cpp

```diff
@@ -1165,28 +1174,35 @@ namespace move_base {
       //we need to set some parameters based on what's been passed in to us to maintain backwards compatibility
       ros::NodeHandle n("~");
       n.setParam("conservative_reset/reset_distance", conservative_reset_dist_);
-      n.setParam("aggressive_reset/reset_distance", circumscribed_radius_ * 4);
-
-      //first, we'll load a recovery behavior to clear the costmap
+      n.setParam("move_backwards/distance_backward", move_backwards_distance_);
+      n.setParam("move_backwards/backwards_velocity", move_backwards_velocity_);
+      
+      //first, clear costmap to clean costmap in front of the robot
       boost::shared_ptr<nav_core::RecoveryBehavior> cons_clear(recovery_loader_.createInstance("clear_costmap_recovery/ClearCostmapRecovery"));
       cons_clear->initialize("conservative_reset", &tf_, planner_costmap_ros_, controller_costmap_ros_);
       recovery_behaviors_.push_back(cons_clear);
 
-      //next, we'll load a recovery behavior to rotate in place
+      //second, we'll load a recovery behavior to move backwards
+      boost::shared_ptr<nav_core::RecoveryBehavior> move_backwards(recovery_loader_.createInstance("move_backwards_recovery/MoveBackRecovery"));
+      move_backwards->initialize("move_backwards", &tf_, planner_costmap_ros_, controller_costmap_ros_);
+      if(move_backwards_enabled_){
+        recovery_behaviors_.push_back(move_backwards);
+      }
+      
+      //third, we'll load rotate recovery
       boost::shared_ptr<nav_core::RecoveryBehavior> rotate(recovery_loader_.createInstance("rotate_recovery/RotateRecovery"));
+      rotate->initialize("rotate_recovery", &tf_, planner_costmap_ros_, controller_costmap_ros_);
```

Comment:
Removed rotate->initialize.  You may want to do a diff to make sure none of the original recovery behaviors were lost.  We should pass the recovery behaviors into the navigation launch file.